pages/vote.tsx

```diff
@@ -39,43 +38,18 @@ class VotePage extends React.Component<VoteProps, VoteState> {
         Router.replace('/')
       }
     }
-    if (req) {
-      const sessionSubmissionsUrl = process.env.GET_SUBMISSIONS_URL
-      const secure = (req.connection as any).encrypted || req.headers['x-forwarded-proto'] === 'https'
-      const url = sessionSubmissionsUrl.startsWith('/')
-        ? 'http' + (secure ? 's' : '') + '://' + req.headers.host + sessionSubmissionsUrl
-        : sessionSubmissionsUrl
-      const response = await fetch(url)
-
-      if (response.status !== 200) {
-        return {}
-      }
-
-      const body = await response.json()
-      return { sessions: body }
-    }
-    return {}
   }
 
   componentWillMount() {
     this.setState({
       isError: false,
-      isLoading: !this.props.sessions,
-      sessions: this.props.sessions || [],
+      isLoading: true,
+      sessions: [],
     })
   }
 
   componentDidMount() {
-    if (this.props.sessions) {
-      this.setSessions(this.props.sessions)
-      return
-    }
-
     const that = this
-    this.setState({
-      isError: false,
-      isLoading: true,
-    })
     fetch(this.props.pageMetadata.appConfig.getSubmissionsUrl)
       .then(response => {
         if (response.status !== 200) {
```

Comment:
Should we use `response.ok` instead?  originally was looking for sttus code because I forgot about `ok`? Minor and can be done in a separate pr...

Reply:
Fair. Yes we should. I must confess I was only looking at shifting the code around rather than doing any tidying up. I do this in a future PR.